packages/xo-server-auth-ldap/src/index.js

```diff
@@ -112,6 +118,69 @@ Or something like this if you also want to filter by group:
       type: 'string',
       default: DEFAULTS.filter,
     },
+    users: {
+      title: 'Users',
+      description: 'Configure how LDAP users are added to XO',
+      type: 'object',
+      properties: {
+        idAttribute: {
+          title: 'ID attribute',
+          description:
+            'Attribute used to identify a user. Must be unique. e.g.: `dn`',
+          type: 'string',
+          default: DEFAULTS.users.idAttribute,
+        },
+      },
+    },
+    groups: {
+      title: 'Synchronize groups',
+      description: 'Import groups from LDAP directory',
+      type: 'object',
+      properties: {
+        synchronize: {
+          title: 'Enable',
+          description:
+            "Import (or update) LDAP groups automatically every time a user logs into XO. If you don't enable this, you can still synchronize the LDAP groups manually from the Settings > Groups page.",
+          type: 'boolean',
+          default: DEFAULTS.groups.synchronize,
+        },
+        base: {
+          title: 'Base',
+          description: 'Where to look for the groups.',
+          type: 'string',
+        },
+        filter: {
+          title: 'Filter',
+          description:
+            'Filter used to find the groups. e.g.: `(objectClass=groupOfNames)`',
+          type: 'string',
+        },
+        idAttribute: {
+          title: 'ID attribute',
+          description:
+            'Attribute used to identify a group. Must be unique. e.g.: `gid`',
+          type: 'string',
+        },
+        displayNameAttribute: {
+          title: 'Display name attribute',
+          description:
+            "Attribute used to determine the group's name in XO. e.g.: `cn`",
+          type: 'string',
+        },
+        displayNamePrefix: {
+          title: 'Display name prefix',
+          description:
+            'Optionally set a prefix for group names in XO imported from the LDAP directory. e.g.: `LDAP-`',
+          type: 'string',
+        },
+        membersAttribute: {
+          title: 'Members attribute',
+          description:
+            'Attribute used to find the members of a group. e.g.: `memberUid`. The values must reference the user IDs (cf. user ID attribute)',
+          type: 'string',
+        },
```

Comment:
I think we need to add a setting for the attribute used for the mapping on the user side.

Maybe with an intermediary object:

```
memberMapping: { groupAttribute = 'memberUid', userAttribute = 'uidNumber' }
```